Innowise.Clinic.Auth.IntegrationTesting/TokenGenerationTests.cs

```diff
@@ -0,0 +1,286 @@
+using System.IdentityModel.Tokens.Jwt;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Json;
+using System.Text;
+using System.Threading.Tasks;
+using Innowise.Clinic.Auth.Dto;
+using Innowise.Clinic.Auth.Jwt;
+using Microsoft.AspNetCore.Mvc.Testing;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using Microsoft.IdentityModel.Tokens;
+using Xunit;
+
+namespace Innowise.Clinic.Auth.IntegrationTesting;
+
+public class TokenGenerationTests : IClassFixture<IntegrationTestingWebApplicationFactory>
+{
+    private readonly IntegrationTestingWebApplicationFactory _factory;
+    private readonly HttpClient _httpClient;
+    private readonly IOptions<JwtData> _jwtData;
+
+    public TokenGenerationTests(IntegrationTestingWebApplicationFactory factory)
+    {
+        _factory = factory;
+        _httpClient = factory.CreateClient(new WebApplicationFactoryClientOptions()
+        {
+            AllowAutoRedirect = false
+        });
+
+        _jwtData = _factory.Services.GetService<IOptions<JwtData>>() ?? throw new InvalidOperationException();
+    }
+
+
+    [Fact]
+    public async Task TestRegisteredUserWithValidEmailAndPasswordGetsValidTokens_OK()
+    {
+        // Arrange
+
+        var validUserRegistrationData = new PatientCredentialsDto()
+        {
+            Email = $"test{TestHelper.UniqueNumber}@test.gmail.com",
+            Password = "12345678"
+        };
+
+        // Act
+
+        var response = await _httpClient.PostAsJsonAsync(TestHelper.SignUpEndpointUri, validUserRegistrationData);
+
+        // Assert
+
+        Assert.True(response.IsSuccessStatusCode);
+
+        var generatedTokens = await response.Content.ReadFromJsonAsync<AuthTokenPairDto>();
+
+        Assert.NotNull(generatedTokens);
+        Assert.NotNull(generatedTokens.SecurityToken);
+        Assert.NotNull(generatedTokens.RefreshToken);
+
+        var userId = TestHelper.ExtractUserIdFromJwtToken(generatedTokens.SecurityToken);
+        var refreshTokenId = TestHelper.ExtractRefreshTokenId(generatedTokens.RefreshToken);
+
+
+        Assert.True(_factory.UseDbContext(x => x.RefreshTokens
+            .Any(rt => rt.TokenId == refreshTokenId && rt.UserId == userId)));
+    }
+
+    [Fact]
+    public async Task TestExpiredJwtTokenRefreshedWithValidRefreshToken_OK()
+    {
+        // Arrange
+
+        var validEmail = $"test{TestHelper.UniqueNumber}@test.gmail.com";
+
+        var userRegistrationDataWithRegisteredEmail = new PatientCredentialsDto()
+        {
+            Email = validEmail,
+            Password = "12345678"
+        };
+
+        // Act
+
+        var response =
+            await _httpClient.PostAsJsonAsync(TestHelper.SignUpEndpointUri, userRegistrationDataWithRegisteredEmail);
+        var generatedTokens = await response.Content.ReadFromJsonAsync<AuthTokenPairDto>();
+
+        await Task.Delay(_jwtData.Value.TokenValidityInSeconds * 1000);
```

Comment:
if you configure validity to smth like 60 minutes the test will be executing forever :).